plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/grouped-product-selector/product-item-template/edit.tsx

```diff
@@ -87,32 +87,53 @@ export default function ProductItemTemplateEdit(
 	} );
 
 	const { product } = useProductDataContext();
+	const [ products, setProducts ] = useState< ProductResponseItem[] | null >(
+		null
+	);
 
-	const { products } = useSelect(
-		( select ) => {
-			const postType = 'product';
-			const { getEntityRecords } = select( coreStore );
+	useEffect( () => {
+		const fetchChildProducts = async ( groupedProductIds: number[] ) => {
+			if ( ! groupedProductIds || groupedProductIds.length === 0 ) {
+				return;
+			}
 
-			let query: Record< string, unknown > = {
-				per_page: 3,
-			};
+			resolveSelect( productsStore )
+				.getProducts( {
+					include: groupedProductIds,
+					per_page: groupedProductIds.length,
+				} )
```

Comment:
We could query only for the `id` here to save some bytes. :slightly_smiling_face: 

```suggestion
				.getProducts( {
					include: groupedProductIds,
					per_page: groupedProductIds.length,
					_fields: [ 'id' ],
				} )
```

Reply:
Nice, thanks!